src/CoreWCF.NetTcp/tests/ServiceBehaviorUseSynchronizationContextAsyncTests.cs

```diff
@@ -0,0 +1,169 @@
+            TaskService.s_synchronizationContextWhenCalled = null;
+            TaskService.s_managedThreadIdWhenCalled = null;
+
+            TaskServiceWithUseSynchronizationContextFalse.s_synchronizationContextWhenCreated = null;
+            TaskServiceWithUseSynchronizationContextFalse.s_synchronizationContextWhenCalled = null;
+        }
+
+        /// <summary>
+        /// This test exists only to demonstrate what the UIFact attribute does.
+        /// </summary>
+        [UIFact]
+        public async Task UIFactTestAsync()
+        {
+            int originalThread = Environment.CurrentManagedThreadId;
+            await Task.Yield();
+            Assert.Equal(originalThread, Environment.CurrentManagedThreadId);
+        }
+
+        [UIFact]
+        public async Task UseSynchronizationContextTrueWithSynchronizationContextTestAsync()
+        {
+            SynchronizationContext orgSynchronizationContext = SynchronizationContext.Current;
+            Assert.NotNull(orgSynchronizationContext); // Set by the UIFact attribute
+            int orgManagedThreadId = Environment.CurrentManagedThreadId;
+
+            await RunAsync<TaskService>();
+
+            Assert.Same(orgSynchronizationContext, TaskService.s_synchronizationContextWhenCreated);
+            Assert.Same(orgSynchronizationContext, TaskService.s_synchronizationContextWhenCalled);
+            Assert.Equal(orgManagedThreadId, TaskService.s_managedThreadIdWhenCalled);
+        }
+
+        [Fact]
+        public async Task UseSynchronizationContextTrueWithoutSynchronizationContextTestAsync()
+        {
+            await RunWithoutSynchronizationContextAsync(async () =>
+            {
+                Assert.Null(SynchronizationContext.Current);
+                int orgManagedThreadId = Environment.CurrentManagedThreadId;
+
+                await RunAsync<TaskService>();
+
+                Assert.Null(TaskService.s_synchronizationContextWhenCreated);
+                Assert.Null(TaskService.s_synchronizationContextWhenCalled);
+            });
+        }
+
+        [UIFact]
+        public async Task UseSynchronizationContextFalseWithSynchronizationContextTestAsync()
+        {
+            Assert.NotNull(SynchronizationContext.Current); // Set by the UIFact attribute
+
+            await RunAsync<TaskServiceWithUseSynchronizationContextFalse>();
+
+            Assert.Null(TaskServiceWithUseSynchronizationContextFalse.s_synchronizationContextWhenCreated);
+            Assert.Null(TaskServiceWithUseSynchronizationContextFalse.s_synchronizationContextWhenCalled);
+        }
+
+        [Fact]
+        public async Task UseSynchronizationContextFalseWithoutSynchronizationContextTestAsync()
+        {
+            await RunWithoutSynchronizationContextAsync(async () =>
+            {
+                Assert.Null(SynchronizationContext.Current);
+                int orgManagedThreadId = Environment.CurrentManagedThreadId;
+
```

Comment:
Same comment as above about getting the current thread id.

Reply:
Same mistake here